src/components/panel/panel.spec.js

```diff
@@ -1222,6 +1299,69 @@ describe('$mdPanel', function() {
           panel2.close();
           panel3.close();
         });
+
+    it('should close the first open panel of any group that the panel is in ' +
+        'when more than the maxium number of panels is opened', function() {
+          $mdPanel.newPanelGroup('test1', {
+            maxOpen: 2
+          });
+          $mdPanel.newPanelGroup('test2', {
+            maxOpen: 3
+          });
+
+          var config1 = {
+            groupName: ['test1', 'test2']
+          };
+          var config2 = {
+            groupName: 'test1'
+          };
+          var config3 = {
+            groupName: 'test2'
+          };
+
+          var panel1 = $mdPanel.create(config1);
+          var panel2 = $mdPanel.create(config1);
+          var panel3 = $mdPanel.create(config2);
+          var panel4 = $mdPanel.create(config3);
+          var panel5 = $mdPanel.create(config3);
+          var panel6 = $mdPanel.create(config3);
```

Comment:
This just got way too complicated. How about one group with maxOpen = 1 and one group with maxOpen =2? I don't understand why you need 6 panels to test this, even with the current values.

It would also be useful to name these related to what their config is. Below when you say panel4, I have no idea which config that is using, so it's hard to understand what you are testing and even harder to know whether it is correct.